### ISSUE TITLE
Floating on Mobile - Default Behaviour

### ISSUE THREAD

**Issue by thexmanxyz** [osano/cookieconsent#277]
I think the current behaviour which is imposed by the following lines:

https://github.com/insites/cookieconsent/blob/master/src/cookieconsent.js#L647-L649

```
if (util.isMobile()) {
  positionStyle = 'floating';
}
```

should be changed. It is not expected that the banner switches in mobile mode automatically to floating. This should be an optional parameter to either opt-in or opt-out. My personal preference would be opt-in to float on mobile.

**Comment by Shawn-Cao** [comment]
It surprised me also. I'm using below code to disable the behavior before a config option is available:
`window.cookieconsent.utils.isMobile = () => false;`

**Comment by thexmanxyz** [comment]
@Shawn-Cao That's a pretty elegant fix for the issue, n1 ;).

**Comment by dGilli** [comment]
I don't think this has been fixed and the workaround is still required

**Comment by felipetoffolo1** [comment]
For now i'm putting this on the initialise object

`"onPopupOpen": function(){
        //workaround for issue on ipad
         var floatingEl = document.getElementsByClassName("cc-floating")
         console.log(floatingEl)
         if(floatingEl && floatingEl[0]){
           floatingEl[0].classList.remove("cc-floating")
           floatingEl[0]element.classList.add("cc-banner")
         }
      }`

**Comment by webdev2027** [comment]
@felipetoffolo1  you've missed a `.`in your last code line.
`floatingEl[0].element.classList.add("cc-banner")` - now it works perfect. Thanks.

**Comment by alexmorleyfinch** [comment]
The reason we do this is because banners look really bad on small screens (unless you style it to look nice I suppose). So we defaulted all mobiles to be floating, and tweaked the styles a little so the floating div touches all the edges of the mobile window. This works for most configurations, so it's not something we'd be willing to change.

I'd be interested to see how any of you guys made a `banner` style work on mobile at all tbh... We really struggled haha

**Comment by thexmanxyz** [comment]
@alexmorleyfinch Mhm TBH I don't have an issue with my banner on mobile devices. It looks just fine. The `font-size` might be a problem on mobile devices but that's something that doesn't depends on the actual layout. 

You can check on one of my sites https://fitness-leitl.at which uses the modified version of the banner style.

**Comment by alexmorleyfinch** [comment]
Thanks for the example! To be fair that's not a bad solution. Maybe we could do a flex-wrap on the banner version to stop it from looking shitty, then it would also remain consistent without surprising anybody. This would be something I'll consider in the next release cycle.

**Comment by thexmanxyz** [comment]
@alexmorleyfinch np ;). However I will remove the link if it isn't needed anymore and replace it with a screenshot because it is a clients website. Just wanted that you see it on a live website. This sounds good because it isn't expected that the banner switches like it currently does. Moreover as the issue is already that long open I missed that this only affects devices which arent within the following media query:

```
@media screen and (max-width: 414px) and (orientation: portrait), screen and (max-width: 736px) and (orientation: landscape)
```

basically this means that tablets which are identified as mobile devices will receive the floating style. So smartphoness typically won't get floating style (even if they use the class). That's was the original problem why it thought it is so inconsistent. I just came back into my mind during testing on the [demo site](https://cookieconsent.insites.com/download/)

reopen the issue as it is for a future release as it is still open :)?

**Comment by thexmanxyz** [comment]
Added PR to disable the default "forced" behavior #436

**Comment by thexmanxyz** [comment]
@alexmorleyfinch Added PR with the functionality to disable the default ("forced") behavior - see #436

**Comment by Softmaker-Ltd** [comment]
There should I put this line of code to disable forced floating on mobile devices:

window.cookieconsent.utils.isMobile = () => false;
